### PR TITLE
PLATUI-1892 extend package lock report

### PR DESCRIPTION
**Pull request by ellamdav**
Productionise PoC code to try to resolve Github repos from the commit ids stored against each Heroku slug.  See [google doc](https://docs.google.com/document/d/1uKVOCTHfkj9dKHjKdD9CGcMOJXjlYonCPAS4ShG7xPQ/edit#heading=h.y7cuchf9fbi3) for full writeup.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,12 +105,19 @@ prototype-2	prototype-2-prototype	FALSE
...
+* If the repo has a package.json
+* If the repo has a package-lock.json
+* If the prototype is deployed to Heroku
+* If the prototype is running in Heroku
+* If the Github repo was found
+
+The output will be saved to `package-lock-analysis.tsv`.
 
 ```shell script
 sbt -Dheroku.apiToken=REPLACE_WITH_TOKEN "generateHerokuReport report.txt"
```
Comment (oscarduignan) [inline README.md:123]: @ellamdav does this script need a `-Dgithub.apiToken=REPLACE_WITH_GITHUB_TOKEN` too?

File: src/main/scala/uk/gov/hmrc/initprototype/GithubManager.scala [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+package uk.gov.hmrc.initprototype
+
+import play.api.libs.json._
+import scalaj.http.{Http, HttpResponse}
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future, blocking}
+import scala.util.{Success, Try}
+
+class GithubManager(config: GithubConfiguration)(implicit ec: ExecutionContext) {
```
Comment (oscarduignan) [inline src/main/scala/uk/gov/hmrc/initprototype/GithubManager.scala:26]: @ellamdav @JoPintoPaul what do you think about renaming the managers to connectors, so GithubConnector instead? Not essential, just did a double take and wasn't sure what manager meant, where connector feels like it's go the role of "sends messages to external services"

File: src/main/scala/uk/gov/hmrc/initprototype/HerokuConfiguration.scala [hunk 3]
```diff
@@ -24,7 +24,7 @@ class HerokuConfiguration {
   private val config: Config           = ConfigFactory.load()
   val baseUrl: String                  = config.getString("heroku.baseUrl")
   val apiToken: String                 = config.getString("heroku.apiToken")
-  val timeout: Duration                = Duration(config.getInt("heroku.timeoutMs"), MILLISECONDS)
+  val jobTimeout: Duration             = Duration(config.getInt("heroku.jobTimeoutMs"), MILLISECONDS)
```
Comment (oscarduignan) [inline src/main/scala/uk/gov/hmrc/initprototype/HerokuConfiguration.scala:27]: @ellamdav what about `reportTaskTimeoutMs`?

Comment (oscarduignan) [reply src/main/scala/uk/gov/hmrc/initprototype/HerokuConfiguration.scala:27]: nevermind, seen it's been used in multiple contexts!

**Review by oscarduignan — DISMISSED** [review]
looks good to me from a high level, few comments but nothing essential so approving and leave it ala carte!

File: README.md (hunk 1 above)
Comment (ellamdav) [reply README.md:123]: doh!  yes, good spot - will add

**Review by oscarduignan — APPROVED** [review]

**Review by oscarduignan — APPROVED** [review]
there's a bit extra we might want to do that I hadn't thought about, where some repos with differing names will now have their heroku url in their repository.yaml but I think that should be a future change

**Pull request merged**